src/execution_plan/execution_plan.c

```diff
@@ -619,16 +619,16 @@ static void _ExecutionPlanSegment_ConvertClause(GraphContext *gc, AST *ast, Exec
 	// Because 't' is set using the offsetof() call, it cannot be used in switch statements.
 	if(t == CYPHER_AST_MATCH) {
 		// Only add at most one set of traversals per plan. TODO Revisit and improve this logic.
-		if(plan->root && (ExecutionPlan_LocateFirstOp(plan->root, OPType_NODE_BY_LABEL_SCAN) ||
-						  ExecutionPlan_LocateFirstOp(plan->root, OPType_ALL_NODE_SCAN))) {
+		if(plan->root && ExecutionPlan_LocateOpMatchingType(plan->root, UNOPTIMIZED_SCAN_OPS,
+															UNOPTIMIZED_SCAN_OP_COUNT)) {
 			return;
 		}
 		_ExecutionPlan_ProcessQueryGraph(plan, plan->query_graph, ast, plan->filter_tree);
 	} else if(t == CYPHER_AST_CALL) {
 		_buildCallOp(ast, plan, clause);
 	} else if(t == CYPHER_AST_CREATE) {
 		// Only add at most one Create op per plan. TODO Revisit and improve this logic.
-		if(ExecutionPlan_LocateFirstOp(plan->root, OPType_CREATE)) return;
+		if(plan->root && ExecutionPlan_LocateOp(plan->root, OPType_CREATE)) return;
```

Comment:
How come we didn't checked for `op->root` before?

Reply:
:+1: Good catch, this change was unnecessary!